src-plugins/manualRegistration/manualRegistrationToolBox.cpp

```diff
@@ -332,40 +327,73 @@ void manualRegistrationToolBox::synchroniseMovingFuseView()
-        return;
+    if (d->controller)
+    {
+        if (d->controller->checkLandmarks() != DTK_FAILURE)
+        {
+            int transformChosen = d->transformType->itemData(d->transformType->currentIndex()).toInt();
 
-    if (d->controller->checkLandmarks() == DTK_FAILURE)
-        return;
+            // Check that Affine process has enough landmarks
+            if (transformChosen == manualRegistration::AFFINE)
+            {
+                if ((d->controller->getPoints_Fixed()->count() < 4) || (d->controller->getPoints_Moving()->count() < 4))
+                {
+                    displayMessageError("Affine transformation needs 4 landmarks minimum by dataset");
+                    return;
+                }
+            }
 
-    d->process = new manualRegistration();
-    d->process->SetFixedLandmarks(d->controller->getPoints_Fixed());
-    d->process->SetMovingLandmarks(d->controller->getPoints_Moving());
-    d->process->setFixedInput(qobject_cast<medAbstractLayeredView*>(d->leftContainer->view())->layerData(0));
-    d->process->setMovingInput(qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0));
-    d->process->setParameter(d->transformType->itemData(d->transformType->currentIndex()).toInt());
-    d->process->update(itkProcessRegistration::FLOAT);
+            d->process = dynamic_cast<medAbstractRegistrationProcess*> (dtkAbstractProcessFactory::instance()->create("manualRegistration"));
 
-    medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+            medRegistrationSelectorToolBox* toolbox = dynamic_cast<medRegistrationSelectorToolBox*>(selectorToolBox());
+            if(toolbox) // toolbox empty in Pipelines and not Registration workspace
+            {
+                toolbox->setProcess(d->process);
+            }
 
-    if(toolbox) // toolbox empty in Pipelines and not Registration workspace
-    {
-        toolbox->setProcess(d->process);
+            manualRegistration* process_Registration = dynamic_cast<manualRegistration*>(d->process);
+            process_Registration->SetFixedLandmarks(d->controller->getPoints_Fixed());
+            process_Registration->SetMovingLandmarks(d->controller->getPoints_Moving());
+            process_Registration->setFixedInput(qobject_cast<medAbstractLayeredView*>(d->leftContainer->view())->layerData(0));
+            process_Registration->setMovingInput(qobject_cast<medAbstractLayeredView*>(d->rightContainer->view())->layerData(0));
+            process_Registration->setParameter(transformChosen);
+
+            // Handle progression bar and run registration
+            medRunnableProcess *runProcess = new medRunnableProcess;
+            runProcess->setProcess (d->process);
+
+            d->progressionStack->addJobItem(runProcess, tr("Progress:"));
+            d->progressionStack->setActive(runProcess,true);
+            d->progressionStack->disableCancel(runProcess);
```

Comment:
note for myself: this should be changed after the merge of the new version of https://github.com/Inria-Asclepios/medInria-public/pull/266